### ISSUE TITLE
[Feature Request] Web hooks

### ISSUE THREAD

**Issue by tomcashman** [digitalfondue/lavagna#7]
It would be great if it was possible for Lavagna to listen for and send web hooks based on certain events.

Example use cases:
* In Gitlab/Github/Service Desk software I could configure my project to post a JSON object to a web hook that would automatically create an new work item on a board
* When a work item is moved to a Done column a JSON object could be posted from Lavagna to a configurable URL that would allow the receiver to auto-close an issue.

**Comment by syjer** [comment]
Strongly agree :).

It's something that is planned for a next release.

**Comment by Kickball** [comment]
This feature would make Lavagna a major player (in my opinion anyway).

Look forward to seeing it implemented!

**Comment by syjer** [comment]
If you could provide some additional uses cases it would be cool, as I don't want to miss some very obvious one :)

**Comment by Kickball** [comment]
I'm currently using Taiga, and I'm having some problems integrating with Github, these are the relevant support pages though.

* [Github Intergration](https://taiga.io/support/github-integration/)
* [Webhooks](https://taiga.io/support/webhooks/)
* [More Webhooks](http://taigaio.github.io/taiga-doc/dist/webhooks.html)
* [Taiga Webhook Code](https://github.com/taigaio/taiga-back/tree/master/taiga/hooks)

I believe they support issues and comment creation from Github on Taiga.

**Comment by cfarence** [comment]
@Kickball I've tried Taiga before, had problems integrating with my Gitlab server. Web hooks with this would make it great with a Gitlab server.

**Comment by Kickball** [comment]
Yeah the Taiga hooks are not perfect, but they might be a good basis for Lavagna to work from.

**Comment by syjer** [comment]
Rest assured that we will try to do a good (and flexible!) webhook implementation :)

The work on it will begin after we have upgraded/refactored the authentication system. I'll provide a more clear/defined roadmap.

**Comment by Fonata** [comment]
I'd love to see two-way sync of GitLab issues and lavagna items

**Comment by syjer** [comment]
@Fonata yep, sorry for taking so much time, but it's something that we will do :)
